Add ESLint to repo (as a grunt task)

Testing out my new [**eslint-config-fxa**](https://github.com/pdehaan/eslint-config-fxa/) shareable ESLint config, and noticed a few errors and warnings in this repo:

### Steps to repro:
1. `$ npm i eslint eslint-config-fxa -D`
2. Create an /.eslintrc file with the following magic YAML sauce:
    ```yaml
extends: fxa/server

rules:
      strict: 0
```
2. `$ eslint .`

### Actual results:
```
$ eslint .

Gruntfile.js
   7:6   warning  fs is defined but never used  no-unused-vars
   7:25  warning  Extra semicolon               semi
   8:29  warning  Extra semicolon               semi
   9:48  warning  Extra semicolon               semi
  17:15  warning  Extra semicolon               semi
  19:43  warning  Extra semicolon               semi
  35:5   warning  Extra semicolon               semi
  38:28  warning  Extra semicolon               semi
  50:7   warning  Extra semicolon               semi
  67:9   warning  Extra semicolon               semi
  70:15  warning  Extra semicolon               semi
  71:9   warning  Extra semicolon               semi
  72:7   warning  Extra semicolon               semi
  73:5   warning  Extra semicolon               semi
  75:68  warning  Extra semicolon               semi
  77:2   warning  Extra semicolon               semi

bin/server.js
  45:72  error  "body" is not defined                             no-undef
  71:4   error  Don't use process.exit(); throw an error instead  no-process-exit

mailer.js
  24:4  error  Trailing spaces not allowed  no-trailing-spaces

templates/index.js
  4:35  error  Strings must use singlequote  quotes

test/local/translator_tests.js
  3:33  error  A space is required after ','  comma-spacing

test/mail_helper.js
  24:4   warning  mail is defined but never used  no-unused-vars
  33:16  error    Strings must use singlequote    quotes
  39:12  warning  uid is defined but never used   no-unused-vars

translator.js
  35:64  error  Strings must use singlequote  quotes

✖ 25 problems (7 errors, 18 warnings)
```

### Expected results:
Less errors/warnings. Most are nonsense and single quotes or semi-colon consistency, but the /bin/server.js `body` one looks like a legit bug.

I fixed most of these w/ #46 PR. New output should be:
```
$ eslint .

bin/server.js
  45:72  error  "body" is not defined  no-undef

test/mail_helper.js
  24:4   warning  mail is defined but never used  no-unused-vars
  39:12  warning  uid is defined but never used   no-unused-vars

✖ 3 problems (1 error, 2 warnings)
```

--- a/.eslintrc
+++ b/.eslintrc
@@ -1,0 +1,3 @@
+{
+  "extends": "fxa/server"
+}

--- a/.travis.yml
+++ b/.travis.yml
@@ -15,4 +15,5 @@ before_install:
 
 script:
   - npm run outdated
+  - grunt lint
   - npm test

--- a/Gruntfile.js
+++ b/Gruntfile.js
@@ -4,9 +4,8 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-const fs = require('fs');
-const path = require('path');
-const extract = require('jsxgettext-recursive');
+const path = require('path')
+const extract = require('jsxgettext-recursive')
 
 // # Globbing
 // for performance reasons we're only matching one level down:
@@ -14,9 +13,11 @@ const extract = require('jsxgettext-recursive');
 // use this if you want to recursively match all subfolders:
 // 'test/spec/**/*.js'
 module.exports = function (grunt) {
-  'use strict';
+  'use strict'
 
-  grunt.loadNpmTasks('grunt-contrib-copy');
+  require('load-grunt-tasks')(grunt)
+
+  grunt.loadNpmTasks('grunt-contrib-copy')
 
   grunt.initConfig({
     copy: {
@@ -32,10 +33,10 @@ module.exports = function (grunt) {
         }]
       }
     }
-  });
+  })
 
   grunt.registerTask('l10n-extract', 'Extract strings from templates for localization.', function () {
-    var done = this.async();
+    var done = this.async()
 
     var walker = extract({
       'input-dir': path.join(__dirname, 'templates'),
@@ -47,7 +48,7 @@ module.exports = function (grunt) {
         '.txt': 'handlebars',
         '.html': 'handlebars'
       }
-    });
+    })
 
     walker.on('end', function () {
       var jsWalker = extract({
@@ -64,14 +65,19 @@ module.exports = function (grunt) {
         parsers: {
           '.js': 'javascript'
         }
-      });
+      })
 
       jsWalker.on('end', function () {
-        done();
-      });
-    });
-  });
+        done()
+      })
+    })
+  })
 
-  grunt.registerTask('default', [ 'copy:strings', 'l10n-extract' ]);
+  // load local Grunt tasks
+  grunt.loadTasks('tasks')
 
-};
+  grunt.registerTask('lint', 'Alias for eslint tasks', ['eslint'])
+
+  grunt.registerTask('default', [ 'copy:strings', 'l10n-extract' ])
+
+}

--- a/bin/server.js
+++ b/bin/server.js
@@ -42,7 +42,7 @@ P.all(
           res.send(200)
         }
         else {
-          log.error({ op: 'send', err: { message: 'invalid type', body: body }})
+          log.error({ op: 'send', err: { message: 'invalid type', body: req.body }})
           res.send(400)
         }
         next()

--- a/package.json
+++ b/package.json
@@ -27,10 +27,13 @@
     "restify": "2.8.2"
   },
   "devDependencies": {
+    "eslint-config-fxa": "1.6.0",
     "grunt": "0.4.5",
     "grunt-cli": "0.1.13",
     "grunt-contrib-copy": "0.5.0",
+    "grunt-eslint": "15.0.0",
     "jsxgettext-recursive": "0.0.5",
+    "load-grunt-tasks": "3.2.0",
     "mailparser": "0.4.4",
     "simplesmtp": "0.3.32",
     "tap": "0.4.12"

--- a/tasks/eslint.js
+++ b/tasks/eslint.js
@@ -1,0 +1,16 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+module.exports = function (grunt) {
+  'use strict'
+
+  grunt.config('eslint', {
+    options: {
+      eslintrc: '.eslintrc'
+    },
+    app: [
+      '*.js', 'bin/*.js', 'tasks/*.js', 'templates/*.js', 'test/**/*.js'
+    ]
+  })
+}

--- a/templates/index.js
+++ b/templates/index.js
@@ -1,7 +1,7 @@
 var path = require('path')
 var P = require('bluebird')
 var handlebars = require('handlebars')
-var readFile = P.promisify(require("fs").readFile)
+var readFile = P.promisify(require('fs').readFile)
 
 handlebars.registerHelper(
   't',

--- a/test/local/translator_tests.js
+++ b/test/local/translator_tests.js
@@ -1,6 +1,6 @@
 var test = require('tap').test
 
-require('../../translator')(['en','pt_br', 'DE', 'ES_AR', 'ES_cl'], 'en')
+require('../../translator')(['en', 'pt_br', 'DE', 'ES_AR', 'ES_cl'], 'en')
 .done(
   function (translator) {
     test(

--- a/test/mail_helper.js
+++ b/test/mail_helper.js
@@ -2,7 +2,6 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-var Mail = require('lazysmtp').Mail
 var MailParser = require('mailparser').MailParser
 var restify = require('restify')
 var config = require('rc')(
@@ -21,7 +20,6 @@ var config = require('rc')(
 
 // SMTP half
 
-var mail = new Mail(config.mail.host)
 var users = {}
 
 function emailName(emailAddress) {
@@ -30,13 +28,12 @@ function emailName(emailAddress) {
 
 require('simplesmtp').createSimpleServer(
   {
-    SMTPBanner: "FXATEST"
+    SMTPBanner: 'FXATEST'
   },
   function (req) {
     var mp = new MailParser({ defaultCharset: 'utf8' })
     mp.on('end',
       function (mail) {
-        var uid = mail.headers['x-uid']
         var link = mail.headers['x-link']
         var rc = mail.headers['x-recovery-code']
         var vc = mail.headers['x-verify-code']

--- a/translator.js
+++ b/translator.js
@@ -32,7 +32,7 @@ module.exports = function (locales, defaultLanguage) {
       var supportedLanguages = []
       for (var i = 0; i < translations.length; i++) {
         var t = translations[i]
-        var language = i18n.normalizeLanguage(i18n.languageFrom(t.locale_data.messages[""].lang))
+        var language = i18n.normalizeLanguage(i18n.languageFrom(t.locale_data.messages[''].lang))
         supportedLanguages.push(language)
         var translator = new Jed(t)
         translator.language = language

